usdt-impl/src/common.rs

```diff
@@ -8,26 +8,45 @@ use crate::DataType;
 
 // Construct function call that is used internally in the UDST-generated macros, to allow
 // compile-time type checking of the lambda arguments.
-pub fn generate_type_check(types: &[DataType]) -> TokenStream {
+pub fn generate_type_check(
+    provider_name: &str,
+    probe_name: &str,
+    types: &[DataType],
+) -> TokenStream {
+    // If the probe has zero arguments, verify that the result of calling the closure is `()`
+    if types.is_empty() {
+        return quote! {
```

Comment:
I think if we passed in the closure object or the span for the closure we could use `quote_spanned!` to have the error point at the closure more specifically.

Reply:
as discussed: I was wrong about its applicability here in that we're generating macros and the errors come from those macros

Reply:
Ok, so I think this won't apply here. We're generating a macro_rules! macro from this code, and we don't have the span associated with the argument closure that the caller ultimately passes in here. We'll have to make do with this for now, so I'll resolve this and the related comments. Thanks for the review!